noxfile.py

```diff
@@ -0,0 +1,261 @@
+        "tests",
+        *session.posargs,
+        silent=True,
+    )
+
+
+@nox.session(name="unit-tests", reuse_venv=True, python=["3.9", "3.10", "3.11", "3.12"])
+def unit_tests(session: Session) -> None:
+    """Run unit tests and generate coverage report"""
+    _install_requirements(session)
+    session.run(
+        "pytest",
+        "--log-level=DEBUG",
+        "--cov=cachi2",
+        "--cov-config=pyproject.toml",
+        "--cov-report=term",
+        "--cov-report=html",
+        "tests/unit",
+        *session.posargs,
+    )
+
+
+@nox.session(name="coverage-html", reuse_venv=True)
+def coverage_html(session: Session) -> None:
+    """Open the coverage report in the browser"""
+    session.install("coverage")
+    session.run("coverage", "html")
+    session.run("xdg-open", "htmlcov/index.html", external=True)
+
+
+def _run_integration_tests(session: Session, env: dict[str, str]) -> None:
+    _install_requirements(session)
+    default_env = {
+        "CACHI2_TEST_NETRC_CONTENT": "machine 127.0.0.1 login cachi2-user password cachi2-pass",
+        "CACHI2_TEST_LOCAL_PYPISERVER": "true",
+        "CACHI2_TEST_LOCAL_DNF_SERVER": "true",
+    }
+    default_env.update(env)
+    session.run(
+        "pytest",
+        "--log-cli-level=WARNING",
+        "tests/integration",
+        *session.posargs,
+        env=default_env,
+    )
+
+
+@nox.session(name="integration-tests", reuse_venv=True)
+def integration_tests(session: Session) -> None:
+    """Run integration tests only for the affected code base in the current branch"""
+    _run_integration_tests(
+        session,
+        {},
+    )
+
+
+@nox.session(name="all-integration-tests", reuse_venv=True)
+def all_integration_tests(session: Session) -> None:
+    """Run all integration tests that are available"""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_RUN_ALL_INTEGRATION_TESTS": "true",
+        },
+    )
+
+
+@nox.session(name="generate-test-data", reuse_venv=True)
+def generate_test_data(session: Session) -> None:
+    """Run all integration tests that are available and update SBOMs"""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_RUN_ALL_INTEGRATION_TESTS": "true",
+            "CACHI2_GENERATE_TEST_DATA": "true",
+        },
+    )
+
+
+@nox.session(name="build-image", reuse_venv=True)
+def build_image(session: Session) -> None:
+    """Build the cachi2 container image"""
+    session.run(
+        "podman",
+        "build",
+        "--tag",
+        "localhost/cachi2:latest",
+        ".",
+        external=True,
+    )
+
+
+@nox.session(name="build-pristine-image", reuse_venv=True)
+def build_pristine_image(session: Session) -> None:
+    """Build the cachi2 container image from scratch"""
+    session.run(
+        "podman",
+        "build",
+        "--pull-always",
+        "--no-cache",
+        "--tag",
+        "localhost/cachi2:latest",
+        ".",
+        external=True,
+    )
```

Comment:
I wonder if anyone's ever used this feature. If one suspects a problem originating in an old base image then `--pull-always` and `--no-cache` should be in their docker arsenal. I think we can safely drop this, it was just a weird kludge that doesn't really relate to the project itself in any way.
That said, since this existed as a Makefile target, it should be dropped either prior this conversion or after, not in the same commit, the migration should be a 1:1 migration of Makefile + tox, maybe even split this particular commit in 2 - first migrate tox.ini to nox, then Makefile (or vice versa), then remove the corresponding origins.

Reply:
> Such targets add clues to how cachi2 could be tested. I suggest keeping them in place until there is a superb troubleshooting document.

I don't think this is useful in any way, we're pretty much the only project I've encountered that tried to wrap *some* of podman's flags to do this, anyone even remotely familiar with podman would not really use this. I mean, I think with this @slimreaper35's effort it's time that we drop some of these kludges altogether and only define what's necessary for the project to decrease clutter, improve transparency, promote flexibility in a way where every user is motivated to use standard invocation mechanisms and be in full control over the configuration of such an invocation if what I just wrote makes sense at all. In this very case I don't think any extensive documentation is needed at all, having a pristine image is just one of the troubleshooting and debugging approaches, we're not wrapping other common invocations.